packages/block-library/src/block/test/edit.native.js

```diff
@@ -58,11 +58,77 @@ afterAll( () => {
 } );
 
 describe( 'Reusable block', () => {
+	it( 'inserts a reusable block', async () => {
+		// We have to use different ids because entities are cached in memory.
+		const reusableBlockMock1 = getMockedReusableBlock( 1 );
+		const reusableBlockMock2 = getMockedReusableBlock( 2 );
+
+		// Return mocked responses for the block endpoints.
+		fetchRequest.mockImplementation( ( { path } ) => {
+			if ( path.startsWith( '/wp/v2/blocks?' ) ) {
+				return Promise.resolve( [
+					reusableBlockMock1,
+					reusableBlockMock2,
+				] );
+			} else if ( path.startsWith( '/wp/v2/blocks/1' ) ) {
+				return reusableBlockMock1;
+			} else if ( path.startsWith( '/wp/v2/blocks/2' ) ) {
+				return reusableBlockMock2;
```

Comment:
Why is only the first return a `Promise`, while the other two are synchronous returns? 

There is also another `fetchRequest` mock that conditionally returns a `Promise`. I wonder if we should instead always return a `Promise` but make its body conditionally empty. WDYT?

Reply:
Good catch! Indeed all of them should return a `Promise`, I'll update the `fetchRequest` mocks of this test.

Reply:
I've addressed this issue in [this commit](https://github.com/WordPress/gutenberg/pull/32506/commits/69f42f2aa1359f1df7003cdaa5399ce7bda13cf2).

Reply:
Heads up that I finally removed the `reusableBlockMock2` response because it wasn't required in the test.